### PR TITLE
Fix modules getting removed during syntaxTreeModify

### PR DESCRIPTION
**Pull request by kaje94**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by NipunaMarcus — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java [hunk 1]
```diff
@@ -187,12 +187,14 @@ public void visit(ModulePartNode modulePartNode) {
     public void visit(ImportDeclarationNode importDeclarationNode) {
         addUnusedImportNode(importDeclarationNode);
 
-        if (importDeclarationNode.moduleName().size() > 0 && importDeclarationNode.moduleName().get(0) != null) {
+        if (importDeclarationNode.moduleName().size() > 0 && importDeclarationNode.moduleName()
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java:190]: `importDeclarationNode.moduleName()` please get this out to a variable and use it.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java [hunk 2]
```diff
@@ -187,12 +187,14 @@ public void visit(ModulePartNode modulePartNode) {
...
+                .get(importDeclarationNode.moduleName().size() - 1) != null) {
```
Comment (NipunaMarcus) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java:191]: i don't think you have to check null here as you already check for the `size > 0`

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java (hunk 1 above)
Comment (kaje94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java]: Updated as requested

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java (hunk 2 above)
Comment (kaje94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java]: Agreed, Updated as requested

**Comment by pubudu91** [comment]
We may not need this if we are merging #31560 since the release branch will be merged back to master once beta2 is released.

**Comment by kaje94** [comment]
Closing this as https://github.com/ballerina-platform/ballerina-lang/pull/31560 is merged

**Pull request closed without merging**